tests/yam/migration/zypper_migration.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Perform online migration by zypper migration.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "opensusebasetest";
+use grub_utils 'grub_test';
+use testapi;
+use power_action_utils 'power_action';
+use utils qw(zypper_call reconnect_mgmt_console upload_folders);
+
+sub run {
+    my $self = shift;
+    select_console 'root-console';
+    assert_script_run("echo 'url: " . get_var('SCC_URL') . "' > /etc/SUSEConnect");
+    my $zypper_migration_target = qr/\[num\/q\]/m;
+    my $zypper_continue = qr/^Continue\? \[y/m;
+    script_run("(zypper migration; echo ZYPPER-DONE) |& tee /dev/$serialdev", 0);
+    my $migration_checks = [
+        $zypper_migration_target, $zypper_continue,
+    ];
+    my $out = wait_serial($migration_checks, 600);
+    while ($out) {
+        diag "out=$out";
+        if ($out =~ $zypper_migration_target) {
+            my $target_version = get_var("TARGET_VERSION", get_required_var("VERSION"));
+            $target_version =~ s/-/ /;
+            if ($out =~ /(\d+)\s+\|\s?SUSE Linux.*?$target_version/m) {
```

Comment:
we can interpolate here without intermediate variable.

Reply:
updated, thanks for review.